Assets/Scripts/Actor Components/Buff.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Photon.Pun;
+
+public class Buff : MonoBehaviour
+{
+    [Header("Combat Changes")]
+
+    [SerializeField] Combat combat;
+    [SerializeField] private LayerMask buffedTargetLayer;
+    [SerializeField] private LayerMask defaultTargetLayer;
+
+    [Header("Visual Changes")]
+
+    [SerializeField] private SpriteRenderer spriteRenderer;
+    [SerializeField] private Color buffedColor;
+    [SerializeField] private Color defaultColor;
+
+    [Space (10)]
+
+    [SerializeField] private PhotonView photonView;
+
+    public bool IsBuffed { get; set; }
+
+    public void ApplyBuff()
+    {
+        photonView.RPC("RPC_ApplyBuff", RpcTarget.All);
+    }
+
+    public void RemoveBuff()
+    {
+        photonView.RPC("RPC_RemoveBuff", RpcTarget.All);
+    }
+
+    [PunRPC]
+    private void RPC_ApplyBuff()
+    {
+        combat.AttackEffectLayer = buffedTargetLayer;
+        spriteRenderer.color = buffedColor;
+        IsBuffed = true;
+    }
+
+    [PunRPC]
+    private void RPC_RemoveBuff()
+    {
+        combat.AttackEffectLayer = defaultTargetLayer;
+        spriteRenderer.color = defaultColor;
+        IsBuffed = false;
```

Comment:
Instead of setting the default target layer and color in the inspector and using them here, it might be better to store the actor's target layer and sprite color when applying the buff and then use the stored values here. This would save us from having to edit the layer/color on both the combat/sprite and buff components in the inspector, should we ever change the defaults in the future.